src/onnx/parse_matmul.cpp

```diff
@@ -89,55 +182,84 @@ struct parse_matmul : op_parser<parse_matmul>
         {
             auto s0_lens        = a0->get_shape().lens();
             auto s1_lens        = a1->get_shape().lens();
-            instruction_ref ba0 = a0;
-            instruction_ref ba1 = a1;
-            // try broadcasting if dimensions other than last two do not match
-            if(not std::equal(
-                   s0_lens.rbegin() + 2, s0_lens.rend(), s1_lens.rbegin() + 2, s1_lens.rend()))
+
+            if(not is_quant_dot and args.size() > 2)
+            {
+                MIGRAPHX_THROW("PARSE_MATMUL: Bias Args not supported for MatMul");
+            }
+
+            bool has_ba0        = false;
+            bool has_ba1        = false;
+            instruction_ref ba0 = set_bias_arg(args, 2, a0, has_ba0);
+            instruction_ref ba1 = set_bias_arg(args, 3, a1, has_ba1);
+
+            // Only INT8 or UINT8 type currently supported
+            std::set<migraphx::shape::type_t> supported_types = {migraphx::shape::uint8_type,
+                                                                 migraphx::shape::int8_type};
+            const auto a0_type                                = a0->get_shape().type();
+            const auto a1_type                                = a1->get_shape().type();
+            const auto ba0_type                               = ba0->get_shape().type();
+            const auto ba1_type                               = ba1->get_shape().type();
+
+            if(is_quant_dot and
+               (not contains(supported_types, ba0_type) or
+                not contains(supported_types, ba1_type) or not contains(supported_types, a0_type) or
+                not contains(supported_types, a1_type)))
```

Comment:
You only need to check for valid types for `a1` and `a0` . biases will have same type as a0 and a1 for which there is a check in set_bias_arg

Reply:
Removed